hazelcast/src/main/java/com/hazelcast/partition/impl/InternalPartitionServiceImpl.java

```diff
@@ -434,6 +434,10 @@ private void promoteFromBackups(Address deadAddress, Address thisAddress) {
                         .setValidateTarget(false)
                         .setService(this);
                 nodeEngine.getOperationService().executeOperation(op);
+
+                MigrationInfo migrationInfo = new MigrationInfo(partition.getPartitionId(), null, partition.getOwnerOrNull());
+                sendMigrationEvent(migrationInfo, MigrationStatus.STARTED);
+                sendMigrationEvent(migrationInfo, MigrationStatus.COMPLETED);
```

Comment:
I think these migration events should be fired in `PromoteFromBackupOperation`, since `executeOperation(op)` is an async call. Events should be ordered with actual operation.